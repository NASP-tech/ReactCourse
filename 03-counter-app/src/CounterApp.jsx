import { useState } from 'react';
import PropTypes from 'prop-types';

const CounterApp = ({ value }) => {

    // console.log('Render');

    const[counter, setCounter] = useState(value);

    const handleAdd = () => {
        setCounter((c) => c + 1);
    }

    const handleSubstract = () => {
        setCounter((c) => c - 1);
    }

    const handleReset = () => {
        setCounter(value);
    }

    return (
        <>
            <h1>CounterApp</h1>
            <h2>{counter}</h2>
            <button onClick={handleAdd}>
                +1
            </button>
            <button onClick={handleSubstract}>
                -1
            </button>
            <button 
                aria-label="btn-reset"
                onClick={handleReset}>
                Reset
            </button>
        </>
    )
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}

export default CounterApp;