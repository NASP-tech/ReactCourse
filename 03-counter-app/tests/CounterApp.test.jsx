import { fireEvent ,render, screen } from "@testing-library/react";
import CounterApp from "../src/CounterApp";

describe('Pruebas en <CounterApp/>', () => { 
    const initialValue = 10;

    test('Debe de hacer match con snapshot', () => { 
        const {container} = render(<CounterApp value={initialValue}/>)
        expect(container).toMatchSnapshot();
    });

    test('debe de mostrar el valor inicial de 100 <CounterApp value={100}/>', () => { 
        render(<CounterApp value={100}/>);
        expect(screen.getByText('100')).toBeTruthy();

        expect(screen.getByRole('heading', {level: 2}).innerHTML).toContain('100');
    });

    test('debe de incrementar con el bóton +1', () => { 
        render(<CounterApp value={initialValue}/>);
        fireEvent.click(screen.getByText('+1'));
        expect(screen.getByText('11')).toBeTruthy();
    })

    test('debe de decrementar con el bóton -1', () => { 
        render(<CounterApp value={initialValue}/>);
        fireEvent.click(screen.getByText('-1'));
        expect(screen.getByText('9')).toBeTruthy();
    })

    test('debe de funcionar el botón de reset', () => { 
        render(<CounterApp value={355}/>);
        fireEvent.click(screen.getByText('+1'));
        fireEvent.click(screen.getByText('+1'));
        fireEvent.click(screen.getByText('Reset'));
        // screen.debug();
        fireEvent.click(screen.getByRole('button', {name: 'btn-reset'}));
        expect(screen.getByText(355)).toBeTruthy();
    })
})  