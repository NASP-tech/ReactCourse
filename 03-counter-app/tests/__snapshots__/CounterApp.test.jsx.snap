// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <CounterApp/> Debe de hacer match con snapshot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
    10
  </h2>
  <button>
    +1
  </button>
  <button>
    -1
  </button>
  <button
    aria-label="btn-reset"
  >
    Reset
  </button>
</div>
`;
